source/frontend/src/components/common/form/FileDragAndDrop.tsx

```diff
@@ -0,0 +1,152 @@
+import { ChangeEvent, FunctionComponent, useState } from 'react';
+import styled from 'styled-components';
+
+import { SectionField } from '../Section/SectionField';
+
+interface IFileDragAndDropProps {
+  onSelectFile: (file: File | null) => void;
+  selectedFile: File | null;
+  validExtensions: string[];
+}
+
+const FileDragAndDrop: FunctionComponent<React.PropsWithChildren<IFileDragAndDropProps>> = ({
```

Comment:
The test coverage on this isn't great. Unit testing this completely may not be realistic but is there anything we can do here?